libraries/c_sdk/standard/mqtt/README.md

```diff
@@ -1,18 +1,71 @@
+
+1. **coreMQTT library** : MQTT library for backward compatibility for MQTT V2.x.x APIs is implemented using [coreMQTT library](https://github.com/FreeRTOS/coreMQTT/blob/master/README.md). Memory estimates for [coreMQTT library](https://github.com/FreeRTOS/coreMQTT/blob/master/README.md) can be found in the documentation [here](https://freertos.org/mqtt/index.html).
+
+2. **Task Pool** : MQTT library for backward compatibility for MQTT V2.x.x APIs depends on [Task Pool](../common/taskpool/). Refer to the table below for calculated code size of Task Pool.
+
+| File | With -O1 Optimization | With -Os Optimization |
+| :-: | :-: | :-: |
+| iot_taskpool.c | 3.8K | 3.0K |
+| **Total estimate** | **3.8K** | **3.0K** |
+
+Please note that the Task Pool may not be exclusively used by this MQTT library for backward compatibility, but may be shared by other libraries. However, the Task Pool is not used by any of the [redesigned libraries for LTS](https://www.freertos.org/ltsroadmap.html).
+
+3. **Platform Abstraction** :  MQTT library for backward compatibility for MQTT V2.x.x APIs depends on the implementation of Platform Abstraction for Network, Clock and Threads. The code size is calculated for the [FreeRTOS implementation of Abstraction Layer](../../../abstractions/platform/freertos) and is added in the table below.
+
+| File | With -O1 Optimization | With -Os Optimization |
+| :-: | :-: | :-: |
+| iot_clock_freertos.c | 0.5K | 0.5K |
+| iot_network_freertos.c | 1.2K | 1.1K |
+| iot_threads_freertos.c | 0.7K | 0.7K |
+| **Total estimate** | **2.4K** | **2.3K** |
+
+Please note that the implementation for Platform Abstraction may not be exclusively used by this MQTT library for backward compatibility, but may be shared by other libraries. However, this Platform Abstraction is not used by any of the [redesigned libraries for LTS](https://www.freertos.org/ltsroadmap.html).
+
+4. **Linear Containers** : MQTT library for backward compatibility for MQTT V2.x.x APIs depends on Linear Containers. However, this implementation is in [c header file](../common/include/iot_linear_containers.h) and the code sizes are already accounted for in the calculated sizes of this library for backward compatibility.
+
+## Tasks created by dependencies of MQTT library for backward compatibility for MQTT V2.x.x APIs
+
+MQTT library for backward compatibility for MQTT V2.x.x APIs depends on Task Pool and Network Interface implementation(Part of Platform Abstraction). Both Task Pool and Network Interface implementation for FreeRTOS creates additional tasks. Task Pool creates `IotTaskPoolInfo.minThreads` number of tasks. Network Interface implementation for FreeRTOS creates a task to receive from the network. Please note these tasks are required in addition to the main application task needed for the application using MQTT library for backward compatibility.
+
+
```

Comment:
I feel we should add total memory size comparison at the end. Customers may not read everything and may not click through all links.  At the top level we can say something like - use of shim require X including all dependencies, use of coreMQTT will require total Y amount

Reply:
Ok. Yea. That will be a good addition. Will add that as well. Thanks